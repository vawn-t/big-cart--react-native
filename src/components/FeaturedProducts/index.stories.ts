import type { Meta, StoryObj } from '@storybook/react-native';

import FeaturedProducts from '.';

const meta: Meta<typeof FeaturedProducts> = {
  title: 'Components/FeaturedProducts',
  component: FeaturedProducts,
};

export default meta;

type Story = StoryObj<React.ComponentProps<typeof FeaturedProducts>>;

export const Base: Story = {
  args: {},
};
