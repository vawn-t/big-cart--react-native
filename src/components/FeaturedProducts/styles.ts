import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  titleWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});

export default styles;
