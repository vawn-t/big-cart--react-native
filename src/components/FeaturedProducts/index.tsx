import { useCallback, useEffect } from 'react';
import { Alert, View } from 'react-native';

// Components
import { LoadingIndicator, Typography } from '@components/common';
import { RightArrow } from '@components/Icon';
import HorizontalCards from '@components/Cards/HorizontalCards';

// Types
import { FontWeight, SizeType, TypoVariant } from '@interfaces/index';

// Constants
import { ALERT } from '@constants/index';

// Hooks
import useProductList from '@hooks/useProduct/useProductList';

// Stores
import useStore from '@stores/useStore';

// Styles
import styles from './styles';

const FeaturedProducts = () => {
  const { data, isSuccess, isPending } = useProductList();

  // Stores
  const setProducts = useStore.use.setProducts();
  const products = useStore.use.products();

  useEffect(() => {
    if (isSuccess) {
      setProducts(data);
    }
  }, [isSuccess]);

  const handleAlertMessage = useCallback(() => {
    Alert.alert(ALERT.COMING_SOON);
  }, []);

  return (
    <>
      <View style={styles.titleWrapper}>
        <Typography
          text="Featured Products"
          variant={TypoVariant.Paragraph2}
          size={SizeType.Large}
          fontWeight={FontWeight.SemiBold}
        />
        <RightArrow onPress={handleAlertMessage} />
      </View>
      {isPending ? (
        <LoadingIndicator />
      ) : (
        <HorizontalCards products={products} />
      )}
    </>
  );
};

export default FeaturedProducts;
