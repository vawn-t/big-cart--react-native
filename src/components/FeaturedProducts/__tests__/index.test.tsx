import FeaturedProducts from '../index';
import { render, screen } from '@utils/index';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

const setup = () => {
  return render(<FeaturedProducts />);
};

describe('FeaturedProducts component', () => {
  it('renders correctly', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the correct title', () => {
    const mockedNavigate = jest.fn();

    jest.mock('@react-navigation/native', () => {
      const actualNav = jest.requireActual('@react-navigation/native');
      return {
        ...actualNav,
        useNavigation: () => ({
          navigate: mockedNavigate,
        }),
      };
    });

    setup();

    expect(screen.getByText('Featured Products')).toBeTruthy();
  });
});
