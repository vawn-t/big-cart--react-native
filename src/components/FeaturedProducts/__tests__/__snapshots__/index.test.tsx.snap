// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedProducts component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#000000",
                  },
                  {
                    "fontFamily": "Poppins-SemiBold",
                  },
                  {
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              Featured Products
            </Text>
            <RNSVGSvgView
              bbHeight="18"
              bbWidth="11"
              fill="none"
              focusable={false}
              height={18}
              onPress={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              responsible={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 18,
                    "width": 11,
                  },
                ]
              }
              width={11}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="m10.3 9.57-9.008 8.21a.76.76 0 0 1-1.07 0 .75.75 0 0 1 0-1.066L8.686 9 .222 1.286a.75.75 0 0 1 0-1.065.76.76 0 0 1 1.07 0L10.3 8.429a.742.742 0 0 1 .215.57.748.748 0 0 1-.215.572Z"
                  fill={
                    {
                      "payload": 4287006857,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <ActivityIndicator
              color="#AEDC81"
              size="small"
              testID="loading-indicator"
            />
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="test"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
